tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/InstanceDiscoveryManagerTests.cs

```diff
@@ -234,8 +253,46 @@ public async Task NetworkProviderIsCalledLastAsync()
 
             // Assert
             Assert.AreSame(_expectedResult, actualResult, "The known metadata provider should be queried second");
-            _knownMetadataProvider.Received(1).GetMetadata("some_env.com", otherEnvs);
-            _ = _networkMetadataProvider.Received(1).FetchAllDiscoveryMetadataAsync(authorityUri, _testRequestContext);
+            _knownMetadataProvider.Received(1).GetMetadata("some_env.com", otherEnvs, Arg.Any<ICoreLogger>());
+            _ = _networkMetadataProvider.Received(1).GetMetadataAsync(authorityUri, _testRequestContext);
```

Comment:
the _ is going to get the task and this won't be awaited within the bounds of the test.  Is this what you intended?  why are you not awaiting for the result of the call within this test?

Reply:
Normally yes, but with NSubstitute it's different. See details here: https://stackoverflow.com/questions/30996024/check-calls-received-for-async-method/31021430#31021430

Reply:
Can you please add that note as a comment here?  Readability wise this stands out as not directly explained by reading the code and this pointer will be helpful to future people reading this test.

Reply:
It looks like in newer versions of NSubstitute we can also use await for this assertion, so I'll change it to that.